opennav_following_demo/demo.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2024 Open Navigation LLC
+# Copyright (c) 2024 Alberto J. Tudela Roldán
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from geometry_msgs.msg import PoseStamped
+from opennav_following_msgs.action import FollowObject
+import rclpy
+import numpy as np
+from rclpy.node import Node
+from rclpy.action import ActionClient
+
+
+class FollowObjectTester(Node):
+
+    def __init__(self):
+        super().__init__(node_name='follow_object_tester')
+        self.subscription = self.create_subscription(
+            PoseStamped, 'detected_dynamic_pose', self.pose_callback, 10)
+        # self.subscription  # prevent unused variable warning
+        self.follow_action_client = ActionClient(self, FollowObject, 'follow_object')
+        self.first_message_received = False
+
+    def pose_callback(self, msg):
+        """Subscribes to the pose topic and calls the action server with the first pose."""
+        if not self.first_message_received:
+            self.first_message_received = True
+            self.send_goal(msg)
```

Comment:
Kind of what I mean - would it be better if this was handled within the server itself and have a topic that it listens to for the pose to follow?